internal/repo/error/repoerror.go

```diff
@@ -0,0 +1,22 @@
+package error
+
+import (
+	"errors"
+)
+
+type PartialErr struct {
+	err error
+}
+
+func Partial(err error) *PartialErr {
+	return &PartialErr{err: err}
```

Comment:
Not super important here, but for safety I think it would be good to check for nil here
```suggestion
	if err == nil {
		return nil
	}
	return &PartialErr{err: err}
```